schema/spacewalk/common/tables/suseContentProjectSource.sql

```diff
@@ -0,0 +1,46 @@
+--
+-- Copyright (c) 2018 SUSE LLC
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+-- Red Hat trademarks are not licensed under GPLv2. No permission is
+-- granted to use or replicate Red Hat trademarks that are incorporated
+-- in this software or its documentation.
+--
+
+CREATE TABLE suseContentProjectSource(
+    id NUMBER NOT NULL
+        CONSTRAINT suse_ct_prj_src_id_pk PRIMARY KEY,
+
+    type VARCHAR2(16) NOT NULL,
+
+    project_id NUMBER NOT NULL
+        CONSTRAINT suse_ct_prj_src_prjid_fk
+            REFERENCES suseContentProject(id)
+            ON DELETE CASCADE,
+
+    channel_id NUMBER
+        CONSTRAINT suse_ct_prj_src_chanid_fk
+            REFERENCES rhnChannel(id)
+            ON DELETE CASCADE,
```

Comment:
Hmm, what should happen if somebody removes a channels which is source of a Content Project?
Maybe we need to handle it explicit in code and not in DB?

Reply:
In this case I'd say it's the same situation as dissociating the channel from Content Project and removing it. Do we want to prevent this? I'd maybe keep it as is now.

Other case is more interesting: removing a channel which is associated to an Environment Target. User could break their Environments easily. I think we want to prevent this definitely. I'm going to change it there.